auslib/web/admin/swagger/api.yaml

```diff
@@ -108,6 +131,110 @@ parameters:
     minimum: 0
     required: true
 
+  change_id_param:
+    name: change_id
+    in: path
+    description: Unique id to identify the identify the change
+    type: string
+    required: true
+
+  field_name_param:
+    name: field
+    in: path
+    description: name of DB column whose history is requested
+    type: string
+    required: true
+
+  userPermissionFormDataParam:
+    name: user_permission_request_body
+    in: body
+    description: Request body containing options and data version field values for manipulating user permission values in DB
+    required: true
+    schema:
+      allOf:
+        - $ref: '#/definitions/CSRFModel'
+        - type: object
+          properties:
+            data_version:
+              type: integer
+              format: int32
+              description: data_version of permission
+              minimum: 1
+              example: 1
+            options:
+              type: ["string", "null"]
+              format: JSONStringField
+              description: JSON field specifying the actions that user can perform and the prdouct on which it has current permission to do so
+              example: "{'actions':['modify']}"
+          # accomodate 'options_as_json' and 'permission' which are duplicate values of 'options'
+          # and 'permission path parameter' respectively, getting passed in request body
+          # No need to validate the duplicate values again
+          additionalProperties: true
+
+
+responses:
+  updateExistingPermission:
```

Comment:
I'm still not sure I understand why this needs to be specific to permissions. You're using generic terms for parameters like change_id, why not here? If you change the descirption to say something like "data version of the updated object", this can be re-used for rule, release, scheduled change, etc. updates.